### PR TITLE
Progress towards storing task/executor state in thread-local storage

### PR DESCRIPTION
**Pull request by rjmccall**

### CODE REVIEW

**Comment by rjmccall** [comment]
@swift-ci Please test

**Comment by rjmccall** [comment]
@swift-ci Please test

**Comment by compnerd** [comment]
I think that this change may have further regressed the Windows CI: https://ci-external.swift.org/job/oss-swift-windows-x86_64-vs2019/4429/console

@shahmishal - seems that the CI didn't trigger here?

File: stdlib/public/Concurrency/Actor.cpp [hunk 1]
```diff
@@ -32,6 +33,144 @@ static bool shouldYieldThread() {
...
+void swift::swift_job_run(Job *job, ExecutorRef executor) {
+  ExecutorTrackingInfo trackingInfo;
+  trackingInfo.enterAndShadow(executor);
+
+  runJobInExecutorContext(job, executor);
+
+  trackingInfo.leave();
+}
+
+void swift::runJobInExecutorContext(Job *job, ExecutorRef executor) {
```
Comment (ktoso) [inline stdlib/public/Concurrency/Actor.cpp:146]: Ok, awesome 👍

**Pull request merged**